src/wire_msg.rs

```diff
@@ -49,30 +66,21 @@ impl WireMsg {
         let dst_length = msg_header.user_dst_len() as usize;
         let payload_length = msg_header.user_payload_len() as usize;
 
-        let mut header_data = BytesMut::with_capacity(header_length);
-        let mut dst_data = BytesMut::with_capacity(dst_length);
-        let mut payload_data = BytesMut::with_capacity(payload_length);
-        // buffer capacity does not actually give us length, so this sets us up
-        header_data.resize(header_length, 0);
-        dst_data.resize(dst_length, 0);
-        payload_data.resize(payload_length, 0);
+        // Check we have all the data and we weren't cut short, otherwise
+        // the following would panic...
+        if !bytes.len() < (header_length + dst_length + payload_length) {
```

Comment:
Shouldn't we compare exact length? ...if not then cleaner if we use `>=` instead of negating `<` comparison?

Reply:
...I'm not sure actually if the condition is correct, using `>=` doesn't sound like what we want to either....?.....